Automattic/wp-calypso
# Reader: remove the requestFeedDiscovery data getter

Replaces the `requestFeedDiscovery` data getter with an imperative React Query query. It continues to have all the desired properties: requests are deduplicated and cached locally in memory (I disabled the IndexedDB persistence for this one).

Then it removes the no-longer-used `waitForHttpData` function, and also an unused `AsyncLoader` component (which mentions `waitForHttpData` in docs example).

**How to test:**
1. Go to `/read/feeds/wptavern.com`
2. Verify that it's redirected to `/read/feeds/124772731`, where the feed ID was retrieved from a REST endpoint.
